core/src/main/scala/org/http4s/headers/Forwarded.scala

```diff
@@ -0,0 +1,121 @@
+      case object Unknown extends Name
+
+      def apply(address: Uri.Ipv4Address): Name = Name.Ipv4(address)
+      def apply(address: Inet4Address): Name = apply(Uri.Ipv4Address.fromInet4Address(address))
+      def apply(a: Byte, b: Byte, c: Byte, d: Byte): Name = apply(Uri.Ipv4Address(a, b, c, d))
+
+      def apply(address: Uri.Ipv6Address): Name = Name.Ipv6(address)
+      def apply(address: Inet6Address): Name = apply(Uri.Ipv6Address.fromInet6Address(address))
+      def apply(a: Short, b: Short, c: Short, d: Short, e: Short, f: Short, g: Short, h: Short)
+          : Name = apply(Uri.Ipv6Address(a, b, c, d, e, f, g, h))
+    }
+
+    sealed trait Port extends Any
+
+    object Port {
+      final case class Num(value: Int) extends AnyVal with Port
+
+      def apply(num: Int): Port = Port.Num(num)
+      def unapply(port: Port): Option[Int] =
+        PartialFunction.condOpt(port) {
+          case Num(num) => num
+        }
+    }
+
+    final case class Obfuscated(value: String) extends AnyVal with Name with Port
+
+    def fromString(s: String): ParseResult[Node] = new NodeParser(s).parse
+  }
+
+  final case class Host(host: Uri.Host, port: Option[Int] = None)
+
+  object Host {
+    def apply(host: Uri.Host, port: Int): Host = apply(host, Some(port))
+
+    def fromString(s: String): ParseResult[Host] = new HostParser(s).parse
+  }
+
+  type Proto = Uri.Scheme
+
+  sealed trait Element extends Any with Product {
+    def `by`: Option[Node]
+    def `for`: Option[Node]
+    def `host`: Option[Host]
+    def `proto`: Option[Proto]
+
+    def `by=`(value: Node): Element
+    def `for=`(value: Node): Element
+    def `host=`(value: Host): Element
+    def `proto=`(value: Proto): Element
+  }
+
+  /**
+    * Enables the following construction syntax (while preserving type safety and consistency):
+    * {{{
+    *   Element.`by=`(<by-node>).`for=`(<for-node>).`host=`(<host>).`proto=`(<schema>)`
+    * }}}
+    */
+  object Element {
+    // Since at least one of the fields must be set to `Some`,
+    // the `Element` trait implementation is hidden.
+    private[this] final case class C(
+        `by`: Option[Node] = None,
+        `for`: Option[Node] = None,
+        `host`: Option[Host] = None,
+        `proto`: Option[Proto] = None)
+        extends Element {
+
+      def `by=`(value: Node): Element = copy(`by` = Some(value))
+      def `for=`(value: Node): Element = copy(`for` = Some(value))
+      def `host=`(value: Host): Element = copy(`host` = Some(value))
+      def `proto=`(value: Proto): Element = copy(`proto` = Some(value))
+
+      override def productPrefix: String = "Element"
+    }
+
+    def `by=`(value: Node): Element = C(`by` = Some(value))
+    def `for=`(value: Node): Element = C(`for` = Some(value))
+    def `host=`(value: Host): Element = C(`host` = Some(value))
+    def `proto=`(value: Proto): Element = C(`proto` = Some(value))
+  }
+
+  type ElementNel = NonEmptyList[Element]
```

Comment:
I've grown regretful over a lot of our type aliases over the years.  Is anything motivating this one?

Reply:
Just traces of my experiments with the model... I'll remove it.